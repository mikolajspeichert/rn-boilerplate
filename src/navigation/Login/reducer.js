import { NavigationActions } from 'react-navigation'

import actions from '@navigation/actions'
import { LoginNavigator } from '.'

const initialState = LoginNavigator.router.getStateForAction(
  LoginNavigator.router.getActionForPathAndParams('Login')
)

export default (state = initialState, action) => {
  switch (action.type) {
    case actions.login.MOVE:
      return LoginNavigator.router.getStateForAction(
        NavigationActions.navigate({ routeName: action.value }),
        state
      )
    default:
      return LoginNavigator.router.getStateForAction(action, state)
  }
}
