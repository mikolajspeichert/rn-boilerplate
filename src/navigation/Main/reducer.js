import { NavigationActions } from 'react-navigation'
import { MainNavigator } from '.'

const initialState = MainNavigator.router.getStateForAction(
  MainNavigator.router.getActionForPathAndParams('Home')
)

export default (state = initialState, action) => {
  switch (action.type) {
    case 'Add':
      return MainNavigator.router.getStateForAction(
        NavigationActions.navigate({ routeName: action.type }),
        state
      )
    default:
      return MainNavigator.router.getStateForAction(action, state)
  }
}
