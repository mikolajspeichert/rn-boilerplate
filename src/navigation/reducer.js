import { combineReducers } from 'redux'
import login from './Login/reducer'
import main from './Main/reducer'

export default combineReducers({
  login,
  main,
})
