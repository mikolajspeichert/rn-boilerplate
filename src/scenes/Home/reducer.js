var initialState = {
  items: [],
  theme: '#fafafa',
}

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state
  }
}
