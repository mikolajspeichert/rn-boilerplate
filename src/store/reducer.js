import { combineReducers } from 'redux'
import { NavigationActions } from 'react-navigation'
import { Navigator } from '@navigation/main'
// import { home } from ''

const initialState = Navigator.router.getStateForAction(
  Navigator.router.getActionForPathAndParams('Home')
)

const nav = (state = initialState, action) => {
  switch (action.type) {
    case 'More':
      return Navigator.router.getStateForAction(
        NavigationActions.navigate({ routeName: 'More' }),
        state
      )
    default:
      return Navigator.router.getStateForAction(action, state)
  }
}

export default combineReducers({
  nav,
})
// export default combineReducers({
//   home,
//   nav,
// })
