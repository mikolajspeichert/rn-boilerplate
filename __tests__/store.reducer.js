import reducer from '@store/reducer'

describe('Main reducer', () => {
  it('reduces initial state', () => {
    expect(reducer({}, { type: 'None' })).toMatchSnapshot()
  })

  it('reduces navigation', () => {
    expect(reducer({}, { type: 'More' })).toMatchSnapshot()
  })
})
