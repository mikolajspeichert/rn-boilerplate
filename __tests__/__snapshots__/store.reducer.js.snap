// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main reducer reduces initial state 1`] = `
Object {
  "nav": Object {
    "index": 0,
    "routes": Array [
      Object {
        "key": "Init-id-1504022106489-0",
        "routeName": "Home",
        "type": undefined,
      },
    ],
  },
}
`;

exports[`Main reducer reduces navigation 1`] = `
Object {
  "nav": Object {
    "index": 1,
    "routes": Array [
      Object {
        "key": "Init-id-1504022106489-0",
        "routeName": "Home",
        "type": undefined,
      },
      Object {
        "key": "id-1504022106489-1",
        "params": undefined,
        "routeName": "More",
      },
    ],
  },
}
`;
